### PR TITLE
RHDEVDOCS-4271: Added a procedure to export an application in Add vie…

### PR DESCRIPTION
**Pull request by shipsing**
Purpose: To resolve this issue: https://issues.redhat.com/browse/RHDEVDOCS-4271

Aligned team: DevConsole

OCP version this PR applies to: 4.12

**Content for preview:** 
[Download creating applications using developer perspective assembly to see the preview in your browser](https://drive.google.com/file/d/1qw1IxtZ4_g-iHOQv-SwADiQqkKtxPgUp/view?usp=sharing)

 [Download viewing application composition using topology view assembly to see the preview in your browser](https://drive.google.com/file/d/1vQ5ET5rV1R1fD20mtJsEYEasKO4ikDeO/view?usp=sharing)

SME review completed: @debsmita1 

QE review completed: @sanketpathak 

Peer review completed: @jc-berger

### CODE REVIEW

**Comment by shipsing** [comment]
Hello @debsmita1, Could you please help with SME review of this PR?

Thanks

File: modules/odc-exporting-an-application-to-another-project-or-cluster.adoc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+. In the developer perspective, perform one of the following steps:
+.. Navigate to the *+Add* view and click *Export application* in the *Application portability* tile.
+.. Navigate to the *Topology* view and click *Export application* on the right side of the view.
+
+. Click *OK* in the *Export Application* dialog box. A notification opens to confirm that the export of resources from your project has started.
+
+. Optional steps that you might need to perform in the following scenarios:
++
+* If you have started exporting an incorrect application, click  *Export application* -> *Cancel Export*.
+* If your export is paused and you want to restart it, click  *Export application* -> *Restart Export*.
```
Comment (debsmita1) [inline modules/odc-exporting-an-application-to-another-project-or-cluster.adoc:25]:
```suggestion
* If your export is already in progress and you want to start a fresh export, click  *Export application* -> *Restart Export*.
```

**Comment by shipsing** [comment]
Thank you @debsmita1. I have incorporated your suggestion. Please verify the same and confirm so that i can proceed with other rounds of review.

**Comment by shipsing** [comment]
@debsmita1, Could you please provide an ack from your side? Thanks

File: modules/odc-exporting-an-application-to-another-project-or-cluster.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+:_content-type: PROCEDURE
+[id="odc-exporting-an-application-to-another-project-or-cluster_{context}"]
+
+= Exporting an application to another project or cluster
+
+As a developer, you can export your application in the ZIP file format. Based on your needs, import the exported application to another project in the same cluster or a different cluster by using the *Import YAML* option in the *+Add* view. This helps you to reuse your application resources and saves your time.
+
+.Prerequisites
+You have installed the gitops-primer operator from the Operator Hub.
```
Comment (debsmita1) [inline modules/odc-exporting-an-application-to-another-project-or-cluster.adoc:13]: In the topology view, to enable the `Export application` button an application needs to be created. Can you add this point as well?

Comment (shipsing) [reply modules/odc-exporting-an-application-to-another-project-or-cluster.adoc]: Incorporated this point @debsmita1

**Comment by shipsing** [comment]
@sanketpathak, Could you please help with QE review of this PR? Thanks

File: modules/odc-viewing-application-topology.adoc [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 You can navigate to the *Topology* view using the left navigation panel in the *Developer* perspective. After you deploy an application, you are directed automatically to the *Graph view* where you can see the status of the application pods, quickly access the application on a public URL, access the source code to modify it, and see the status of your last build. You can zoom in and out to see more details for a particular application.
 
-The *Topology* view also provides you the option to monitor your applications using the *List* view. Use the *List view* icon (image:odc_list_view_icon.png[title="List view icon"]) to see a list of all your applications and use the *Graph view* icon (image:odc_topology_view_icon.png[title="Topology view icon"]) to switch back to the graph view.
+The *Topology* view provides you the option to monitor your applications using the *List* view. Use the *List view* icon (image:odc_list_view_icon.png[title="List view icon"]) to see a list of all your applications and use the *Graph view* icon (image:odc_topology_view_icon.png[title="Topology view icon"]) to switch back to the graph view.
```
Comment (jc-berger) [inline modules/odc-viewing-application-topology.adoc:6]: I defer to you as you know your users more. But for accessibility and usability, be careful with images. They can be particularly difficult to maintain. 

Reaching out to your content strategist would help make this decision. Again, if you know to keep them, leave this content here as is :)

File: modules/odc-viewing-application-topology.adoc [hunk 4]
```diff
@@ -22,3 +22,5 @@ You can customize the views as required using the following:
 ** *Show* elements based on *Pod Count* or *Labels*
 *** Pod Count: Select to show the number of pods of a component in the component icon.
 *** Labels: Toggle to show or hide the component labels.
+
+The *Topology* view also provides you the *Export application* option to download your application in the ZIP file format. You can then import the downloaded application to another project or cluster. For more details, see _Exporting an application to another project or cluster_ in the _Additional resources_ section.
```
Comment (jc-berger) [inline modules/odc-viewing-application-topology.adoc:26]: Do you want to link to the `Exporting an application to another project or cluster`? Or do you simply want to direct users how to access this doc? 

When referring users to actual docs, I'm used to seeing those docs included as links or xrefs.

File: modules/odc-exporting-an-application-to-another-project-or-cluster.adoc [hunk 5]
```diff
@@ -0,0 +1,31 @@
+// Module included in the following assemblies:
+//
+// * applications/creating_applications/odc-creating-applications-using-developer-perspective.adoc
+
+:_content-type: PROCEDURE
```
[the remark this answers is not in the document]
Comment (shipsing) [reply modules/odc-exporting-an-application-to-another-project-or-cluster.adoc:5]: @jc-berger, The same is mentioned here: https://redhat-documentation.github.io/modular-docs/#module_anchor-and-file-names-concept

However, if your team does not include the module prefixes in file names followed by either a hyphen (-) or an underscore (_), include one of the following variables in each file before the anchor ID:

:_content-type: ASSEMBLY
:_content-type: PROCEDURE
:_content-type: CONCEPT
:_content-type: REFERENCE
:_content-type: SNIPPET

File: modules/odc-exporting-an-application-to-another-project-or-cluster.adoc [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+[id="odc-exporting-an-application-to-another-project-or-cluster_{context}"]
+
+= Exporting an application to another project or cluster
+
+As a developer, you can export your application in the ZIP file format. Based on your needs, import the exported application to another project in the same cluster or a different cluster by using the *Import YAML* option in the *+Add* view. This helps you to reuse your application resources and saves your time.
+
+.Prerequisites
+* You have installed the gitops-primer operator from the Operator Hub.
+* The *Export application* option is disabled in the *Topology* view even after installing the gitops-primer operator. To enable *Export application*, ensure that you have created an application in the *Topology* view.
```
[the remark this answers is not in the document]
Comment (shipsing) [reply modules/odc-exporting-an-application-to-another-project-or-cluster.adoc:14]: @jc-berger The suggestion: You have disabled the Export application option in the Topology view even after installing the gitops-primer operator. (The Export application option is automatically disabled, user does not do it manually).

will so some workaround.

File: modules/odc-viewing-application-topology.adoc (hunk 3 above)
Comment (shipsing) [reply modules/odc-viewing-application-topology.adoc:6]: @jc-berger, I agree with your point of maintenance. This is why i only use screenshots where it is actually needed for assistance. 
Here, we have used inline images to show UI icons, which might be scattered across the view. The graphics actually help readers to locate them. So, leaving the images as they are.

File: modules/odc-viewing-application-topology.adoc (hunk 4 above)
Comment (shipsing) [reply modules/odc-viewing-application-topology.adoc:26]: @jc-berger, I guess cross-refernces or links are not supported in any module. This is why i provided the cross-refernce in the additional resources section of the assembly this module is a part of.

See [applications/odc-viewing-application-composition-using-topology-view.adoc](https://github.com/openshift/openshift-docs/pull/49524/files#diff-ee9adcdc33a181eba8d0ad16724b1fbb4b731fddce7a38422c9144bec1d9e26b) in the Files changed tab.

**Comment by shipsing** [comment]
@jc-berger Thank you for your suggestions. I have incorporated them wherever applicable. Please verify the same and confirm.

**Review by jc-berger — APPROVED** [review]
/lgtm

**Comment by abrennan89** [comment]
Following up with this PR after a Slack conversation with the dev team, it seems this content should actually be CP'd back to 4.10.

**Pull request merged**